src/libsync/discovery.h

```diff
@@ -88,6 +88,16 @@ class ProcessDirectoryJob : public QObject
     /** Start up to nbJobs, return the number of job started; emit finished() when done */
     int processSubJobs(int nbJobs);
 
+    void setUnderEncryptedRoot(bool underEncryptedRoot)
+    {
+        _underEncryptedRoot = underEncryptedRoot;
+    }
+
+    bool underEncryptedRoot() const
```

Comment:
What about reusing the `hasEncryptedAncestor` naming you used in your other PR? That's also the term we used at other places so probably makes sense for consistency.

Reply:
@er-vin I'd argue about that. The problem is when moving folders, most of the time, there is no close ancestor that is encrypted. In 99% of cases, this is going to be a root folder, which, of course, is also an ancestor you are right, but, the algorithm here is completely different. We are not using while loop on the full path and trying to find the closest ancestor. We are just using what's already know in the root folder.

So, to put it simply, I've been considering to use this naming that you are proposing, but, since, that naming comes from a completely different part of the system (**remote jobs**), but we are now inside the **discovery**, and also the fact that, this method simply returns the value of a variable without doing any lookup, I've decided, that having a **hasEncryptedAncestor** just because it is a very close term, but not exactly the same, is going to be confusing.

Reply:
I see. Makes sense. Let's just rename the getter to `isUnderEncryptedRoot` then, in order to follow the Qt convention.

Hmmm... Now that I think of it... what about `isInsideEncryptedTree` ? (optional of course, it's just that under and root kind of trips me up everytime, but might be just me so don't want to force it on you, I leave the decision to you).

Reply:
@er-vin Agreed. Renamed and suqashed.